### ISSUE TITLE
LeNet gives poor accuracy.

### ISSUE THREAD

**Issue by husnainmubarik** [hls-fpga-machine-learning/hls4ml#234]
I tried LeNet with mnist dataset; and the accuracy of hls4ml model is 0.25 whereas accuracy of keras model is 0.96. 
Any idea how can I fix that?
![image](https://user-images.githubusercontent.com/47509684/97817594-04097200-1c63-11eb-99fb-7b5a1e8fa0c3.png)
![image](https://user-images.githubusercontent.com/47509684/97817564-cad10200-1c62-11eb-8e15-60171a7340ab.png)

**Comment by thesps** [comment]
This could happen due to choice of fixed-point data types not working well for your model. Did you try any of the `profiling` or `trace` functionality, or tune data types according to your model? 

Since you mentioned MNIST, I think it's relatively likely that you'll need to use the 'Stable' implementation of the Softmax activation (see #207) for details. You can do this by setting something like `config['LayerName']['softmax']['Strategy'] = 'Stable'` in your hls4ml configuration, though I'm not sure what your final activation layer name is, so replace `'softmax'` with whatever that is.

**Comment by Mmiglio** [comment]
Hi @thesps, I'm facing a similar problem. Looking at `profiling` functionality output I see that I may have problems with the `softmax` layer. 

![Screenshot 2020-11-02 at 12 21 44](https://user-images.githubusercontent.com/26923771/97862665-053fab00-1d06-11eb-9714-78ada907a449.png)
Even after tuning the data types (I'm using `ap_fixed<50,5>` to "cover" the `softmax` activation) I can't get even close to my model performance (0.60 vs 0.97 accuracy). Is there something I can do at training level / after training to mitigate this problem ?

**Comment by thesps** [comment]
@Mmiglio did you also try the 'Stable' softmax (like `config['LayerName']['softmax']['Strategy'] = 'Stable'`) ?
